src/libraries/System.Private.CoreLib/src/System/IndexOfAnyValues/ProbabilisticMap.cs

```diff
@@ -201,15 +306,115 @@ internal static int IndexOfAny<TNegator>(ref uint charMap, ref char searchSpace,
             ref char searchSpaceEnd = ref Unsafe.Add(ref searchSpace, searchSpaceLength);
             ref char cur = ref searchSpace;
 
-            while (!Unsafe.AreSame(ref cur, ref searchSpaceEnd))
+            if (IsVectorizationSupported && typeof(TNegator) == typeof(IndexOfAnyAsciiSearcher.DontNegate) && searchSpaceLength >= 16)
```

Comment:
This is a pretty big body to be aggressively inlined and may actually hurt inlining for the other path.

Should this be moved to separate helper method which has its own inlining, that way it can be more easily dropped as dead code when `TNegator` isn't `DontNegate`?

Reply:
I can move it out, but I plan on reworking how this logic is set up soon anyway.
The fact we're using ProbabilisticMap for the -Except case was a mistake, so this will end up being just one path, without the generic negator.